grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java

```diff
@@ -415,20 +419,24 @@ private void useRoundRobinLists(
         subchannels = Collections.unmodifiableMap(newSubchannelMap);
         break;
       case PICK_FIRST:
-        List<EquivalentAddressGroup> eagList = new ArrayList<>();
+        int numOfEags = newBackendAddrList.size();
+        List<EquivalentAddressGroup> eagList = Arrays.asList(new EquivalentAddressGroup[numOfEags]);
         // Because for PICK_FIRST, we create a single Subchannel for all addresses, we have to
         // attach the tokens to the EAG attributes and use TokenAttachingLoadRecorder to put them on
         // headers.
         //
         // The PICK_FIRST code path doesn't cache Subchannels.
+        int offset = pickFirstIndex;
         for (BackendAddressGroup bag : newBackendAddrList) {
           EquivalentAddressGroup origEag = bag.getAddresses();
           Attributes eagAttrs = origEag.getAttributes();
           if (bag.getToken() != null) {
             eagAttrs = eagAttrs.toBuilder()
                 .set(GrpclbConstants.TOKEN_ATTRIBUTE_KEY, bag.getToken()).build();
           }
-          eagList.add(new EquivalentAddressGroup(origEag.getAddresses(), eagAttrs));
+          eagList.set(
+              offset++ % numOfEags,
```

Comment:
Hmmm... Techincally, `pickFirstIndex` could be `MAX_VALUE` and this would wraparound. Since you already chose to use Math.abs(), you could maybe replace that with something akin to `pickFirstGlobalIndex.getAndIncrement() & 0xFFFF`.

Reply:
Done.